src/keystore.h

```diff
@@ -112,14 +117,15 @@ typedef std::map<
 typedef std::map<
     libzcash::SaplingIncomingViewingKey,
     libzcash::SaplingExtendedFullViewingKey> SaplingFullViewingKeyMap;
-// Only maps from default addresses to ivk, may need to be reworked when adding diversified addresses. 
+// Only maps from default addresses to ivk, may need to be reworked when adding diversified addresses.
 typedef std::map<libzcash::SaplingPaymentAddress, libzcash::SaplingIncomingViewingKey> SaplingIncomingViewingKeyMap;
 
 /** Basic key store, that keeps keys in an address->secret map */
 class CBasicKeyStore : public CKeyStore
 {
 protected:
-    HDSeed hdSeed;
+    std::optional<MnemonicSeed> mnemonicSeed;
+    std::optional<HDSeed> legacySeed;
```

Comment:
The options here are "only MnemonicSeed", "only HDSeed", or "none", correct? Please document that constraint or consider a `std::variant`.